### PR TITLE
add cookbook template

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

File: libs/remix-ui/home-tab/src/lib/components/homeTabGetStarted.tsx [hunk 1]
```diff
@@ -61,14 +61,23 @@ function HomeTabGetStarted({plugin}: HomeTabGetStartedProps) {
     let templateDisplayName = TEMPLATE_NAMES[templateName]
     const metadata = TEMPLATE_METADATA[templateName]
     if (metadata) {
-      await plugin.call('dGitProvider', 'clone', {url: metadata.url, branch: metadata.branch}, templateDisplayName)
-      return
+      if (metadata && metadata.type === 'git') {
```
Comment (Aniket-Engg) [inline libs/remix-ui/home-tab/src/lib/components/homeTabGetStarted.tsx:64]: I think `metadata && ` can be removed from both the plugins

File: libs/remix-ui/workspace/src/lib/utils/constants.ts [hunk 2]
```diff
@@ -87,15 +87,33 @@ export const TEMPLATE_NAMES = {
...
+  },
+  'uniswapV4HookBookMultiSigSwapHook': {
+    type: 'plugin',
+    name: 'cookbookdev',
+    endpoint: 'openPattern',
+    params: ['Uniswap-V4-HookBook-MultiSigSwapHook', true]
   }
 }
+
+export type TemplateType = {
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/utils/constants.ts:112]: Not sure if this is a right file to add a type def

**Comment by Aniket-Engg** [comment]
Screenshot 2024-01-08 at 4 56 49 PM

May be we need to do something with the template name

**Comment by Aniket-Engg** [comment]
editor by default open `src/MULTI_SIG/MultiSigSwapHook.sol` which can not be successfully compiled on Remix

**Comment by yann300** [comment]
> editor by default open `src/MULTI_SIG/MultiSigSwapHook.sol` which can not be successfully compiled on Remix

it should compile after the plugin is updated.

File: libs/remix-ui/home-tab/src/lib/components/homeTabGetStarted.tsx [hunk 3]
```diff
@@ -61,14 +61,23 @@ function HomeTabGetStarted({plugin}: HomeTabGetStartedProps) {
     let templateDisplayName = TEMPLATE_NAMES[templateName]
     const metadata = TEMPLATE_METADATA[templateName]
     if (metadata) {
-      await plugin.call('dGitProvider', 'clone', {url: metadata.url, branch: metadata.branch}, templateDisplayName)
-      return
+      if (metadata.type === 'git') {
+        await plugin.call('dGitProvider', 'clone', {url: metadata.url, branch: metadata.branch}, templateDisplayName)
+      } else if (metadata && metadata.type === 'plugin') {
```
Comment (Aniket-Engg) [inline libs/remix-ui/home-tab/src/lib/components/homeTabGetStarted.tsx:66]: Same for this condition, `metadata && ` can be removed

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**